### ISSUE TITLE
Build fails out of the box due to a missing webpack-cli dev dependency and other missing webpack configs

### ISSUE THREAD

**Issue by SeanKilleen** [OfficeDev/generator-office#389]
Hi all,

I'll submit a PR for this but writing up the issue first.

**Update**: Leaving this issue here for a bit to percolate because locking `webpack-dev-server` to a specific version may be the better route to go anyway.

## Symptoms
It appears that out of the box, the build for add-ins (in this case an Outlook add-in) now fails.

Once I resolved the missing `webpack-cli` project (see #390), I also see that `webpack-dev-server` is only compatible with webpack v4.x. So I update webpack v4.x to accommodate, at which point (understandably) the webpack config breaks due to some changes around the major version bump.

## Causes
Webpack has split out the cli into a `webpack-cli` project, which needs to be added as an additional dev dependency.

## Recreation
On Windows 10 | Node v10.14.1 | npm v6.4.1

* `npm install -g yo generator-office`
* `yo office`
* Select `Office add-in project using React framework`
* Entered name 
* Selected `Outlook` as the target app.
* Generates an app in a sub-folder
* run `npm start`

**Expected behavior**: See local site served

**Actual behavior**: See the below error:

```output
> webpack-dev-server --inline --config config/webpack.dev.js --progress

The CLI moved into a separate package: webpack-cli
Please install 'webpack-cli' in addition to webpack itself to use the CLI
-> When using npm: npm i -D webpack-cli
-> When using yarn: yarn add -D webpack-cli
internal/modules/cjs/loader.js:582
    throw err;
    ^
```

## Steps to fix (I think?)

* [ ] (#389) Add webpack-cli dependency
* [ ] Bump webpack to `^4.0.0`
* [ ] Fix `Error: webpack.optimize.CommonsChunkPlugin has been removed, please use config.optimization.splitChunks instead` 

Also, I think it may be possible to lock in `webpack-dev-server` to the last version that doesn't require webpack v4.x

**Comment by akrantz** [comment]
I'm unclear to which webpack config this refers. Is this for a particular project template? I don't think it is for generator-office itself.

**Comment by SeanKilleen** [comment]
Sorry — this was for a react framework based outlook plugin.

**Comment by richdizz** [comment]
It appears to me that the break was introduced in webpack-dev-server v3.0.0, everything works if you set the version to v2.11.1. However, that introduces a high severity vulnerability. Not sure if that is really that critical given this is for local testing.

**Comment by SeanKilleen** [comment]
Given @richdizz's point, it sounds like the best way forward on this would be to bump to Webpack v4 and update things accordingly.

I may be willing to take a look at this in PR format if folks would be willing to offer some guidance along the way. Let me know if you're interested.

**Comment by TCourtneyOwen** [comment]
Thanks for reporting this issue, @SeanKilleen.  I am seeing the same behavior and will try and get a fix out  today.  It appears to be a problem with the React template in general, and not just specific Outlook.

**Comment by TCourtneyOwen** [comment]
I just created a pull request for this at https://github.com/OfficeDev/generator-office/pull/391

**Comment by TCourtneyOwen** [comment]
This issue has been fixed and a new version of Yo Office (1.1.22) has been published out to NPM
